components/service-operator/internal/object/status.go

```diff
@@ -0,0 +1,91 @@
+// +kubebuilder:object:generate=true
+package object
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type State string
+
+var (
+	ErrorState       State = "ERROR"
+	DeletingState    State = "DELETE_IN_PROGRESS"
+	ReadyState       State = "READY"
+	ReconcilingState State = "RECONCILE_IN_PROGRESS"
+)
+
+// +kubebuilder:object:root=false
+
+// Status is the type shared by most service resources
+type Status struct {
+	// Generic service state
+	State State `json:"state,omitempty"`
+	// AWS specific status
+	AWS AWSStatus `json:"aws,omitempty"`
+}
+
+var _ StatusReader = &Status{}
+var _ StatusWriter = &Status{}
+
+// GetStatus returns the status
+func (s *Status) GetStatus() Status {
```

Comment:
This seems interesting - it returns... itself? Or a new object if it itself is nil?

Reply:
it doesn't return a pointer to itself ... it returns a copy of it's status or an empty status .... this is simply implementing the object.StatusReader interface